atlasdb-config/src/main/java/com/palantir/atlasdb/factory/Leaders.java

```diff
@@ -111,20 +113,23 @@ public static LocalPaxosServices createLocalServices(
         Map<PingableLeader, HostAndPort> otherLeaders = generatePingables(
                 remotePaxosServerSpec.remoteLeaderUris(), sslSocketFactory, userAgent);
 
-        ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder()
-                .setNameFormat("atlas-leaders-%d")
-                .setDaemon(true)
-                .build());
+        InstrumentedExecutorService instrumentedExecutor = new InstrumentedExecutorService(
+                Executors.newCachedThreadPool(new ThreadFactoryBuilder()
+                        .setNameFormat("atlas-leaders-%d")
+                        .setDaemon(true)
+                        .build()),
+                AtlasDbMetrics.getMetricRegistry());
 
-        PaxosProposer proposer = createPaxosProposer(ourLearner, acceptors, learners, config.quorumSize(), executor);
+        PaxosProposer proposer = createPaxosProposer(ourLearner, acceptors, learners, config.quorumSize(),
+                instrumentedExecutor);
```

Comment:
Slightly beyond the scope of this PR, but I'm actually wondering if we should have separate executors for proposing & leader election so that they don't intermingle tasks.